src/passes/RemoveUnusedBrs.cpp

```diff
@@ -906,6 +906,12 @@ struct RemoveUnusedBrs : public WalkerPass<PostWalker<RemoveUnusedBrs>> {
         // further optimizations after this, and those optimizations might even
         // benefit from this improvement.
         auto glb = Type::getGreatestLowerBound(curr->castType, refType);
+        if (!getModule()->features.hasCustomDescriptors() && glb.isExact() &&
+            !curr->castType.isExact()) {
+          // When custom descriptors is not enabled, nontrivial exact casts are
+          // not allowed.
+          glb = glb.with(Inexact);
+        }
```

Comment:
Also, is the check here enough? castType might have been exact, which is valid in the rare cases we do allow exact casts (trivial or nullability) , but the glb might not be similarly allowed?

Reply:
The GLB of an exact type and any other type in the same hierarchy is either the same exact type (if the other type is a supertype of it), or the bottom type otherwise. If the original exact type was valid, then both of these possible results are also valid.

Reply:
Oh, right, because nothing is below exact types, thanks.